packages/grant-explorer/src/features/common/Dropdown.tsx

```diff
@@ -0,0 +1,58 @@
+import { Menu, Transition } from "@headlessui/react";
+import {
+  Children,
+  cloneElement,
+  Fragment,
+  PropsWithChildren,
+  ReactElement,
+} from "react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import tw from "tailwind-styled-components";
+
+export function Dropdown({
+  label,
+  children,
+}: { label?: string } & PropsWithChildren) {
+  return (
+    <Menu as="div" className="md:relative inline-block text-left z-20">
+      <div>
+        <Menu.Button className="inline-flex gap-2 items-center">
+          <span className="text-white py-2">{label}</span>
+          <ChevronDownIcon className="h-5 w-5 text-black" aria-hidden="true" />
+        </Menu.Button>
+      </div>
+      <Transition
+        as={Fragment}
+        enter="transition ease-out duration-100"
+        enterFrom="transform opacity-0 scale-95"
+        enterTo="transform opacity-100 scale-100"
+        leave="transition ease-in duration-75"
+        leaveFrom="transform opacity-100 scale-100"
+        leaveTo="transform opacity-0 scale-95"
+      >
+        <Menu.Items className="absolute w-full md:w-auto p-2 right-0 mt-2 origin-top-right rounded-2xl bg-white shadow-lg">
+          {Children.map(children, (child, i) => (
+            <Menu.Item key={i} as={Fragment}>
+              {({ active }) => cloneElement(child as ReactElement, { active })}
+            </Menu.Item>
+          ))}
+        </Menu.Items>
+      </Transition>
+    </Menu>
+  );
+}
+
+export const DropdownItem = tw("div")<{ active?: boolean }>`
```

Comment:
would rather have a default value of false here rather than an optional param, but leaving up to you.

Reply:
If we set it to required, all the components need to specify:
```tsx
<DropdownItem active={false} ... />
```

Or is it possible to set a default value in tw styled components?

Reply:
True, didn't realize that. Closing.